scripts/lib/CIME/utils.py

```diff
@@ -1224,6 +1224,11 @@ def _get_most_recent_lid_impl(files):
 
     return sorted(results)
 
+def sorted_ls(path):
```

Comment:
The name of this function should indicate that this is sorted by timestamp - maybe something like `ls_sorted_by_mtime`